src/Http/Offer/PriceInfoValidatingRequestBodyParser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Offer;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use CultuurNet\UDB3\Json;
+use CultuurNet\UDB3\Model\Serializer\ValueObject\Price\PriceInfoDenormalizer;
+use CultuurNet\UDB3\Model\ValueObject\Price\PriceInfo;
+use Psr\Http\Message\ServerRequestInterface;
+
+class PriceInfoValidatingRequestBodyParser implements RequestBodyParser
+{
+    public function parse(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $data = $request->getParsedBody();
+        if (!isset($data->priceInfo) || !is_array($data->priceInfo)) {
```

Comment:
Nit: Since this class only gets executed after `JsonSchemaValidatingRequestBodyParser`, the `priceInfo` property should always be an array if present. So this if could be reduced to just:
```
if (!isset($data->priceInfo)) {
    return $request; 
}
```

(In some other RequestBodyParsers we also check the type and if other subproperties are set and the correct type etc because those are called _before_ `JsonSchemaValidatingRequestBodyParser` to transform old/legacy JSON request bodies to the ones expected by the JSON schema.)

Reply:
@JonasVHG this is maybe something that can still be done inside this pull request?

Reply:
This is changed.